src/global-table.ts

```diff
@@ -1,22 +1,88 @@
-import { IResource, RemovalPolicy, Resource, Stack } from 'aws-cdk-lib';
+import { IResource, RemovalPolicy, Resource } from 'aws-cdk-lib';
 import * as ddb from 'aws-cdk-lib/aws-dynamodb';
+import * as iam from 'aws-cdk-lib/aws-iam';
 import { Construct } from 'constructs';
+import * as perms from './perms';
 
 export interface IGlobalTable extends IResource {
+  grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant;
+  grantReadData(grantee: iam.IGrantable): iam.Grant;
+  grantWriteData(grantee: iam.IGrantable): iam.Grant;
+  grantReadWriteData(grantee: iam.IGrantable): iam.Grant;
 }
 
 abstract class GlobalTableBase extends Resource implements IGlobalTable {
+  public abstract readonly tableArn: string;
+  protected readonly regionalArns = new Array<string>();
+  public abstract readonly tableName: string;
+  public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [
+        this.tableArn,
+      ],
+      scope: this,
+    });
+  }
+
+  public grantReadData(identity: iam.IGrantable): iam.Grant {
```

Comment:
The implementation of each `grantXYZ` function should call the `grant()` function.